addons/mrp_account/models/product.py

```diff
@@ -54,7 +54,7 @@ def _compute_average_price(self, qty_invoiced, qty_to_invoice, stock_moves):
             bom_line = move.bom_line_id
             bom_line_data = bom_lines[bom_line]
             bom_line_qty = bom_line_data['qty']
-            value += move.product_id._compute_average_price(qty_invoiced * bom_line_qty, qty_to_invoice * bom_line_qty, move)
+            value += bom_line_qty * move.product_id._compute_average_price(qty_invoiced * bom_line_qty, qty_to_invoice * bom_line_qty, move)
```

Comment:
it seems to be a separated fix

Reply:
It's necessary for issue 2 https://github.com/odoo/odoo/pull/71509#issuecomment-851986068

`_compute_average_price` return just the average price of the bom line product.
Without the multiplication you are ignoring the component quantity so you may need 100 Bolts but compute the cost for just 1